src/device_handle.rs

```diff
@@ -540,10 +540,8 @@ impl<T: UsbContext> DeviceHandle<T> {
     pub fn read_string_descriptor_ascii(&self, index: u8) -> crate::Result<String> {
         let mut buf = Vec::<u8>::with_capacity(255);
 
-        let buf_slice = unsafe { slice::from_raw_parts_mut(buf.as_mut_ptr(), buf.capacity()) };
-
-        let ptr = buf_slice.as_mut_ptr() as *mut c_uchar;
-        let len = buf_slice.len() as i32;
+        let ptr = buf.as_mut_ptr() as *mut c_uchar;
+        let len = buf.len() as i32;
```

Comment:
`with_capacity` will always create a vector with length 0, so this will now always return an empty string. This has caused an issue I noticed while upgrading from 0.5.2: https://github.com/jonas-schievink/jaylink/pull/15#issuecomment-655623760.

Reply:
Ohh... of course we must call the capacity method. I fix this today.